File: pg1/apps/test/python/scrollbar_fun.c
/* Test for PicoGUI Python client library. If Python batch requests
 * (via hold()/flush()) are working, this C version shouldn't be
 * much faster than scrollbar_fun.py
 */

#include <picogui.h>
#include <math.h>

int main(int argc, char **argv) {
  pghandle w[40];
  pghandle app;
  int i;
  float t;
  
  pgInit(argc,argv);
  app = pgRegisterApp(PG_APP_NORMAL,"Scrollbar Fun - C",0);
  
  for (i=0;i<40;i++) {
    w[i] = pgNewWidget(PG_WIDGET_SCROLL,PG_DERIVE_INSIDE,app);
    pgSetWidget(PGDEFAULT,
		PG_WP_SIDE, PG_S_LEFT,
		0);
  }
  
  t = 0;
  while (1) {
    t += 0.04;
    for (i=0;i<40;i++)
      pgSetWidget(w[i],
		  PG_WP_VALUE, (int)(50 + 50 * sin(i*0.2+t)),
		  0);
    pgUpdate();
  }

  return 0;
}
    

File: pg1/apps/test/python/scrollbar_fun.py
#!/usr/bin/python2.2
# Test for PicoGUI Python client library. If Python batch requests
# (via hold()/flush()) are working, this C version shouldn't be
# much faster than scrollbar_fun.py

import PicoGUI, math
app = PicoGUI.Application('Scrollbar Fun - Python')

widgets = []
for i in range(0,40):
    widgets.append(app.addWidget('scroll'))
    widgets[i].number = i
    widgets[i].side = 'left'

t = 0
while 1:
    app.server.hold()
    t += 0.04
    for w in widgets:
        w.value = 50 +50 * math.sin( (w.number*0.2) - t )
    app.server.update()
    app.server.flush()
